Adding sitecertificate resource type to Microsoft.Web RP

gargimitra opened the pull request:
Adding a new resource type "sites/certificates" in Microsoft.Web swagger.

# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [x] Other, please clarify:
    - Adding a new resource type for Microsoft.Web swagger.

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

Gargimitra12300 committed:
Adding sitecertificate resource type to Microsoft.Web RP

Adding a new resource type "sites/certificates" in Microsoft.Web swagger.

Gargimitra12300 committed:
Update ListSiteCertificatesByResourceGroup.json

Gargimitra12300 committed:
Update CommonDefinitions.json

Gargimitra12300 committed:
Update CommonDefinitions.json

Gargimitra12300 committed:
Update CommonDefinitions.json

Gargimitra12300 committed:
Update SiteCertificates.json

Gargimitra12300 committed:
Update SiteCertificates.json

Gargimitra12300 committed:
Update SiteCertificates.json

Gargimitra12300 committed:
Update SiteCertificates.json

Gargimitra12300 committed:
Update SiteCertificates.json

Gargimitra12300 committed:
Update SiteCertificates.json

gargimitra commented:
> @gargimitra please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree [company="Microsoft"]

gargimitra commented:
> @microsoft-github-policy-service agree company="your company"
@microsoft-github-policy-service agree company="Microsoft"

Gargimitra12300 committed:
updated def file

Gargimitra12300 committed:
Revert "updated def file"

This reverts commit 99b4c530ce95959c046c8c73841e0f5f543d57f2.

Gargimitra12300 committed:
Update CommonDefinitions.json

Gargimitra12300 committed:
Revert "Update CommonDefinitions.json"

This reverts commit b99bc9a5fa8db32a616ca31c87eda74aaeea2a96.

Gargimitra12300 committed:
Update CommonDefinitions.json

Gargimitra12300 committed:
Update CommonDefinitions.json

Gargimitra12300 committed:
Update CommonDefinitions.json

Gargimitra12300 committed:
Update CommonDefinitions.json

Gargimitra12300 committed:
Update CommonDefinitions.json

Gargimitra12300 committed:
Revert "Update CommonDefinitions.json"

This reverts commit f3851edb1461cae69cbdbb3c5f7ae8c3339696f4.

Gargimitra12300 committed:
Update CommonDefinitions.json

Gargimitra12300 committed:
Update CommonDefinitions.json

Gargimitra12300 committed:
Update readme.go.md

Gargimitra12300 committed:
Revert "Update readme.go.md"

This reverts commit fd6bfd78ec5fea9ca243e27e48accbec941a9681.

Gargimitra12300 committed:
Update CommonDefinitions.json

Gargimitra12300 committed:
Update readme.go.md

Gargimitra12300 committed:
Update SiteCertificates.json

Gargimitra12300 committed:
Revert "Update readme.go.md"

This reverts commit 890e745780d0beefae29d07fb9e558fe676348d7.

gargimitra commented on specification/web/resource-manager/Microsoft.Web/stable/2024-11-01/CommonDefinitions.json:
```diff
@@ -673,6 +673,311 @@
+            "password": {
+              "description": "Certificate password.",
+              "type": "string",
+              "x-ms-mutability": [
```
This property is not updated as a part of patch call, hence set the x-ms-mutability to read here.

However, I am getting errors at " azure-sdk-for-go" stage. https://github.com/Azure/azure-rest-api-specs/pull/31446/checks?check_run_id=32738411013 
I am not sure how to fix this. Can someone help me here please?

edwin-msft replied:
responded in chat regarding duplicates in the .go.md file

gargimitra replied:
I think using x-ms-secret fixed it.

razvanbadea-msft commented on specification/web/resource-manager/Microsoft.Web/stable/2024-11-01/CommonDefinitions.json:
```diff
@@ -673,6 +673,311 @@
+          "description": "Certificate resource specific properties",
+          "type": "object",
+          "properties": {
+            "password": {
```
>password [](http://example.com/codeflow?start=13&length=8)

mark it as a secret using the x-ms-secret attribute

gargimitra replied:
Done

razvanbadea-msft commented on specification/web/resource-manager/Microsoft.Web/stable/2024-11-01/CommonDefinitions.json:
```diff
@@ -673,6 +673,311 @@
+            },
+            "keyVaultId": {
+              "description": "Key Vault Csm resource Id.",
+              "type": "string"
```
>string [](http://example.com/codeflow?start=23&length=6)

you can add the arm-id format

gargimitra replied:
@razvanbadea-msft thanks for the suggestion. I have updated the format to indicate arm-id. It did trigger the "Breaking Change" check to fail. Can you help with that please?

razvanbadea-msft commented on specification/web/resource-manager/Microsoft.Web/stable/2024-11-01/CommonDefinitions.json:
```diff
@@ -673,6 +673,311 @@
+              "readOnly": true
+            },
+            "keyVaultId": {
+              "description": "Key Vault Csm resource Id.",
```
you should use Azure in the descriptions

gargimitra replied:
Done

razvanbadea-msft commented on specification/web/resource-manager/Microsoft.Web/stable/2024-11-01/CommonDefinitions.json:
```diff
@@ -673,6 +673,311 @@
+            },
+            "serverFarmId": {
+              "description": "Resource ID of the associated App Service plan, formatted as: \"/subscriptions/{subscriptionID}/resourceGroups/{groupName}/providers/Microsoft.Web/serverfarms/{appServicePlanName}\".",
+              "type": "string"
```
>string [](http://example.com/codeflow?start=23&length=6)

you can add the arm-id format with a restricted format

gargimitra replied:
@razvanbadea-msft  thanks for the suggestion. I have updated the format to indicate arm-id. It did trigger the "Breaking Change" check to fail. Can you help with that please?

gargimitra replied:
@razvanbadea-msft  are you the right contact for helping me move this PR forward? If not can you point me to the right person please?

razvanbadea-msft commented:
>     "Certificate": {

if this was moved to common definitions why are you not removing them from here?

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2024-11-01/Certificates.json:310 in 15194a8. [](commit_id = 15194a8e8ace47f09e8fdc8eefdfff17fa660174, deletion_comment = False)

Gargimitra12300 committed:
Addressing feedback

Gargimitra12300 committed:
Update CommonDefinitions.json

Gargimitra12300 committed:
Addressing feedback

gargimitra commented:
> > ```
> > "Certificate": {
> > ```
> 
> if this was moved to common definitions why are you not removing them from here?
> 
> Refers to: specification/web/resource-manager/Microsoft.Web/stable/2024-11-01/Certificates.json:310 in [15194a8](https://github.com/Azure/azure-rest-api-specs/commit/15194a8e8ace47f09e8fdc8eefdfff17fa660174). [](commit_id = [15194a8](https://github.com/Azure/azure-rest-api-specs/commit/15194a8e8ace47f09e8fdc8eefdfff17fa660174), deletion_comment = False)

Removed

gargimitra commented:
@razvanbadea-msft  can you help with the review of the latest iteration please? I am not sure how to resolve the error in the PR checks.

Gargimitra12300 committed:
Update sdk-suppressions.yaml

Adding suppression rule for Azure SDK Go as per suggestion obtained from API review channel.

tadelesh committed:
fix the directive

tadelesh committed:
add more directive

tadelesh committed:
remove suppression

tadelesh committed:
fix breaking for go

tadelesh committed:
fix wrong index

tadelesh committed:
try rename

tadelesh committed:
add more transform to resolve the rename bug
